src/topology/schema.rs

```diff
@@ -289,4 +442,303 @@ mod tests {
+            ),
+            (
+                "multiple inputs, sources with schema",
+                TestCase {
+                    inputs: vec![("source-foo", None), ("source-bar", None)],
+                    sources: IndexMap::from([
+                        (
+                            "source-foo",
+                            vec![Output::default(DataType::all()).with_schema_definition(
+                                Definition::empty().required_field(
+                                    "foo",
+                                    Kind::integer().or_bytes(),
+                                    Some("foo bar"),
+                                ),
+                            )],
+                        ),
+                        (
+                            "source-bar",
+                            vec![Output::default(DataType::all()).with_schema_definition(
+                                Definition::empty().required_field(
+                                    "foo",
+                                    Kind::timestamp(),
+                                    Some("baz qux"),
+                                ),
+                            )],
+                        ),
+                    ]),
+                    transforms: IndexMap::default(),
+                    want: vec![
+                        Definition::empty().required_field(
+                            "foo",
+                            Kind::integer().or_bytes(),
+                            Some("foo bar"),
+                        ),
+                        Definition::empty().required_field(
+                            "foo",
+                            Kind::timestamp(),
+                            Some("baz qux"),
+                        ),
+                    ],
+                },
+            ),
+            (
+                "transform overrides source",
+                TestCase {
+                    inputs: vec![("source-bar", None), ("transform-baz", None)],
+                    sources: IndexMap::from([
+                        (
+                            "source-foo",
+                            vec![Output::default(DataType::all()).with_schema_definition(
+                                Definition::empty().required_field(
+                                    "foo",
+                                    Kind::boolean(),
+                                    Some("foo"),
+                                ),
+                            )],
+                        ),
+                        (
+                            "source-bar",
+                            vec![Output::default(DataType::all()).with_schema_definition(
+                                Definition::empty().required_field(
+                                    "bar",
+                                    Kind::integer(),
+                                    Some("bar"),
+                                ),
+                            )],
+                        ),
```

Comment:
I think I'd remove second pipeline as it doesn't affect the thing to be asserted

Reply:
Does the `source-bar` serve any specific purpose here or can we remove it?

To me it looks like it was copied from the `multiple inputs, sources with schema` test case without being very important here.